### ISSUE TITLE
Behavior of CutSet.mix

### ISSUE THREAD

**Issue by ZuoyunZheng** [lhotse-speech/lhotse#808]
Hi,
first of all, just wanted to appreciate lhotse being a great tool and its lively community.

My problem is that i wish to mix 2 CutSets with some length variations. So i wanted to do something like `mixedcuts = cuts1.mix(cuts2)`. I wish that the mixing is performed for the whole duration for all cuts in `cuts1`. 

However, if i leave the `duration` argument to `None`, only one `Cut` is sampled from `cuts2`. If i need to specify a `duration`, there is no clear way for me other than passing a `duration` that is longer than the max Cut length in `cuts1`, which is not the most elegant.

I somehow wonder, shouldn't it be more intuitive for `CutSet.mix` to perform mixing for the whole duration for each  if `duration` is left `None`?

**Comment by pzelasko** [comment]
Good point. Do you mind making a PR to add this feature?

**Comment by ZuoyunZheng** [comment]
I just opened a PR. I'm pretty new to this, so i most likely would require some guidance.

Just FYI, in `CutMix`, the longest duration from the source `CutSet` is passed as `duration` argument to `mix`. Not sure if it's a design choice or a result of `CutSet.mix`'s behavior.

**Comment by pzelasko** [comment]
I think it's more of a design choice, it's intended to be able to mix the noise into padding areas in addition to speech. I would probably keep it as some users might depend on this behavior. If you need it to do sth else, we can always add an option for it.

**Comment by ZuoyunZheng** [comment]
Then i have a follow up question.
So if i understand correctly, `CutMix` now mixes noise for the max sequence duration of a batch (given by the `CutSet`), including to padding areas for shorter sequences.

However, in neural network forwarding, for example in conformer ctc recipes, we always use a `Supervision` input which contains length information for each sequence in the batch. The forwarding therefore only takes into account the real length of each sequence and not the complete padded length/max sequence length in batch. So i don't see the necessity to pad noise for padding areas, for example for speech recognition's CTC or RNNT loss calculations. 

Please tell me if i'm missing anything. I'm intrigued to hear more.

**Comment by pzelasko** [comment]
You are correct. This feature may be more immediately useful in other tasks that involve some sort of audio segmentation, e.g. speech activity detection or diarization. 

As for ASR applications, we had an idea before about concatenating multiple short utterances for ASR training and having multiple supervisions point to different subsets of a single feature matrix. I think it was difficult to implement with transformer models so in the end Icefall recipes dropped the support for it. You can see how it works e.g. here in CutConcatenate examples: https://github.com/lhotse-speech/lhotse/pull/234
